File: algo-expert/python/algo_expert/arrays/easy/validate_subsequence.py
def isValidSubsequence(array: list, sequence: list):
    if len(sequence) > len(array):
        return False

    i = 0
    for num in sequence:
        while i < len(array) and array[i] != num:
            i += 1
        if i >= len(array):
            return False
        i += 1
    return True

File: algo-expert/typescript/src/arrays/easy/ValidateSubSequence.ts
export function isValidSubsequence(array: number[], sequence: number[]): boolean {
  let j = 0;
  for (let i in sequence) {
    let isInArray = false;
    while (j < array.length) {
      if (sequence[i] === array[j]) {
        array.splice(j, 1);
        isInArray = true;
        break;
      }
      j++;
    }
    if (!isInArray) {
      return false;
    }
  }
  return true;
}

